app/src/main/java/at/vunfer/openrealms/model/DeckGenerator.java

```diff
@@ -101,8 +101,9 @@ private static Card parseCard(XmlPullParser xmlParser)
         return new Card(cardName, cardCost, cardEffects);
     }
 
-    private static void addCardAbility(List<Effect> cardEffects, XmlPullParser xmlParser)
+    private static List<Effect> getCardAbility(XmlPullParser xmlParser)
```

Comment:
also changed this method so it doesn't modify the incoming object.

Not that it's a big problem _here_, but it's cleaner that way

[1](https://stackoverflow.com/questions/12019978/is-it-a-good-practice-to-change-arguments-in-java) [2](https://softwareengineering.stackexchange.com/questions/245767/is-modifying-an-incoming-parameter-an-antipattern)